src/modules/keyboardq/KeyboardQmlViewStep.h

```diff
@@ -0,0 +1,75 @@
+/* === This file is part of Calamares - <https://github.com/calamares> ===
+ *
+ *   Copyright 2014-2015, Teo Mrnjavac <teo@kde.org>
+ *   Copyright 2017, Adriaan de Groot <groot@kde.org>
+ *
+ *   Calamares is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   Calamares is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with Calamares. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef KEYBOARDQMLVIEWSTEP_H
+#define KEYBOARDQMLVIEWSTEP_H
+
+#include <QObject>
+#include "Config.h"
+#include <utils/PluginFactory.h>
+#include <viewpages/QmlViewStep.h>
+
+#include <DllMacro.h>
+#include <QQmlParserStatus>
+
+class KeyboardPage;
+
+class PLUGINDLLEXPORT KeyboardQmlViewStep : public Calamares::QmlViewStep, public QQmlParserStatus
+{
+    Q_INTERFACES(QQmlParserStatus)
+    Q_OBJECT
+
+public:
+    explicit KeyboardQmlViewStep( QObject* parent = nullptr );
+
+    QString prettyName() const override;
+    QString prettyStatus() const override;
+
+    void classBegin() override;
+    void componentComplete() override;
+
+    bool isNextEnabled() const override;
+    bool isBackEnabled() const override;
+
+    bool isAtBeginning() const override;
+    bool isAtEnd() const override;
+
+    QList< Calamares::job_ptr > jobs() const override;
```

Comment:
There's a modern alias for this, see ba8811c62851f0412ac753d8fd1c633f7220ebdb (you couldn't have known about it, I don't think, but it illustrates the kind of proliferating tech-debt we see)